File: backend/src/main/java/com/ezweb/model/ComponentRegistry.java
package com.ezweb.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "component_registry")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ComponentRegistry {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 100)
    private String name;

    @Column(length = 500)
    private String description;

    @Column(name = "s3_file_url", nullable = false, length = 500)
    private String s3FileUrl;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String schema; // JSON schema stored as text

    @Column(length = 50)
    private String category;

    @Column(length = 20)
    private String version = "1.0.0";

    @Column(nullable = false)
    private Boolean active = true;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
}

File: userapp/flutter_application_1/lib/models/component_registry.dart
import 'package:json_annotation/json_annotation.dart';

part 'component_registry.g.dart';

@JsonSerializable()
class ComponentRegistry {
  final int id;
  final String name;
  final String? description;
  final String s3FileUrl;
  final String schema; // JSON string
  final String? category;
  final String version;
  final bool active;
  final DateTime createdAt;

  ComponentRegistry({
    required this.id,
    required this.name,
    this.description,
    required this.s3FileUrl,
    required this.schema,
    this.category,
    required this.version,
    required this.active,
    required this.createdAt,
  });

  factory ComponentRegistry.fromJson(Map<String, dynamic> json) => _$ComponentRegistryFromJson(json);
  Map<String, dynamic> toJson() => _$ComponentRegistryToJson(this);
}
